### PR TITLE
[Mining][RPC] Fix `getnetworkhashps` rpc

### LINKED ISSUE #1018: [Bug][RPC][Mining] The network hashrate returned by `getmininginfo` rpc is wrong
The problem is that the function that should return the network hashrate for one of the three mining algos, calculates the average hashrate using the `nChainWork` _"Total amount of work (expected number of hashes) in the chain up to and including this block"_ which **includes the work of all three algos** and if I see correctly even also PoS work.
https://github.com/Veil-Project/veil/blob/4107c99a2029a07f89ad0b1e0f68c03ec55b7350/src/rpc/mining.cpp#L159

This is also the reason why all the returned network hashrates for the three different algos are always almost the same even though their difficulties vary a lot.

### PR DESCRIPTION
**Pull request by us77ipis**
The hashrates returned by the `getnetworkhashps` rpc were always wrong, because the calculation was based on the entire work of the chain (including all three mining algos and also proof-of-stake).

Fixes #1018

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -70,75 +70,97 @@ static UniValue GetNetworkHashPS(int lookup, int height, std::string type) {
     if (pb == nullptr || !pb->nHeight)
         return 0;
 
+    bool isPoS = type == "pos";
+    bool isSha256D = type == "sha256d";
+    bool isRandomX = type == "randomx";
+    bool isProgPow = type == "progpow";
+    bool isX16RT = type == "xr16t";
```
Comment (Zannick) [inline src/rpc/mining.cpp:77]: Nit: this could probably be turned into an enum value and then used as a switch in the if-else blocks below.

**Review by Zannick — APPROVED** [review]

File: src/rpc/mining.cpp (hunk 1 above)
Comment (us77ipis) [reply src/rpc/mining.cpp]: True, that's the cleaner way, just changed that.

**Review by Zannick — APPROVED** [review]
utACK 15eeef7de68c101715a09a50cc8c85c235b330d5

**Comment by seanPhill** [comment]
I've been running this for some weeks.

**Pull request merged**